### PR TITLE
main: implement --tag-relative=never|always

### PR DESCRIPTION
**Pull request by masatake**
Proposed in #868.

"always" and "never" as an argument for --tag-relative

```

Even if "yes" is specified as an option argument for --tag-relative,
absolute paths are used in tags output if an input is given as
an absolute path. This behavior is expected in exuberant-ctags
as written in its man-page.

In addition to "yes" and "no", universal-ctags takes "never" and "always".

If "never" is given, absolute paths are used in tags output regardless
the path representation for input file(s). If "always" is given,
relative paths are used always.

Signed-off-by: Masatake YAMATO <yamato@redhat.com>
```

### CODE REVIEW

**Comment by blueyed** [comment]
In the commit message: s/regardless the/regardless of the/

File: docs/news.rst [hunk 1]
```diff
@@ -349,6 +349,21 @@ For specifying only one pseudo tag, specify it without sign:
...
+
+Even if "yes" is specified as an option argument for --tag-relative,
+absolute paths are used in tags output if an input is given as
+an absolute path. This behavior is expected in exuberant-ctags
+as written in its man-page.
+
+In addition to "yes" and "no", universal-ctags takes "never" and "always".
+
+If "never" is given, absolute paths are used in tags output regardless
+the path representation for input file(s). If "always" is given,
```
Comment (blueyed) [inline docs/news.rst]: s/regardless the/regardless of the/

File: main/options.c [hunk 2]
```diff
@@ -367,8 +367,10 @@ static optionDescription LongOptionDescription [] = {
  {1,"       Define regular expression for locating tags in specific language."},
  {0,"  --sort=[yes|no|foldcase]"},
  {0,"       Should tags be sorted (optionally ignoring case) [yes]?"},
- {0,"  --tag-relative=[yes|no]"},
+ {0,"  --tag-relative=[yes|no|always|never]"},
  {0,"       Should paths be relative to location of tag file [no; yes when -e]?"},
+ {0,"       always: be relative even if input files are passed in absolute paths" },
+ {0,"       never:  be absolute even if input files are passed in relative paths" },
```
Comment (blueyed) [inline main/options.c]: s/passed in/passed in with/

**Comment by masatake** [comment]
@blueyed, thank you for comments.

**Comment by blueyed** [comment]
@masatake 
I assume I should test and merge it?  Or at least provide feedback?

**Comment by masatake** [comment]
@blueyed, I would like you to test my change.
I'm trying to understand the result of the test case on "Environment: compiler=msys2, MSYS2_ARCH=x86_64, MSYS2_DIR=msys64, MSYSTEM=MINGW64".

**Comment by masatake** [comment]
It seems that absoluteFilename and relativeFilename on the platform don't work as we expected.

**Comment by masatake** [comment]
@k-takata san, I need your help.

As far as seeing the test result it seems that relativeFilename and absoluteFilename are not implemented well on environment "Environment: compiler=msys2, MSYS2_ARCH=x86_64, MSYS2_DIR=msys64, MSYSTEM=MINGW64". I would like to correct the implementation of
these two functions on the platform.

I got Windows-8.1 (kvm guest on Fedora), and I installed msys2-64. `autogen.sh && ./configure && make ` works fine. I got ctags.exe for the branch I proposed in this PR. Surprisingly the test case failed in approver is passed successfully on my environment. I wonder what is the difference between my environment and msys2 of appveyor. Any idea?

**Comment by masatake** [comment]
I have found the reason why my code doesn't work on msys2: both `/` and `\\` are used in a path.

**Comment by masatake** [comment]
On msys2 the output of pwd of bash is different from the value provided by getcwd function.

**Comment by masatake** [comment]
The runtime environment is the same between bash where pwd is called and ctags which calls getcwd.

**Comment by masatake** [comment]
I will add code do as follows:
On the platform where separators used in getcwd function and pwd command are different, `--tag-relative=never` and --tag-relative=always` are disabled.

**Comment by masatake** [comment]
@blueyed, do you use windows?
I decide to disable `--tag-relative=never|always` on Windows. Is it o.k.?

**Comment by masatake** [comment]
@k-takata,  in the environment "Environment: compiler=msys2, MSYS2_ARCH=x86_64, MSYS2_DIR=msys64, MSYSTEM=MINGW64" , pwd command used in a test harness script and getcwd function called in ctags return paths with different separators. I don't have actual output of the two now.

I introduced window8.1 and set up a msys2 environment for reproducing what I see in locally, but I could not  get the same result.

**Comment by masatake** [comment]
I will merge this after merging #1230.

**Comment by blueyed** [comment]
@masatake 
I am not using Windows - sorry for the late reply.

Thanks a lot for getting this in!

**Pull request closed without merging**